### PR TITLE
Monitor multiple masters with single sentinel

### LINKED ISSUE #73: Support configuring multiple sentinel commands
Firstly, thanks for this cookbook!

We usually overload sentinels to watch multiple master servers.  I've hacked this together by overriding the existing attributes, template, and provider, but before I clean it up and submit a pull request, I wanted to see if you had plans for this and if so, how you intended it.

I ask mainly because there is a *monitor* default attribute that isn't used anywhere, and I haven't found any configuration in redis that corresponds to that keyword aside from  *sentinel monitor*.

I've basically rolled up the sentinel config options into an array for the monitor attribute.

Sample config (testing on a single, vagrant instance):

    sentinels: [
        # Primary Sentinel
        {
          'sentinel_port' => 26379,
          'name' => '26379',
          'monitor' => [
              {'master_name'  => 'redis1',
                'master_ip'   => '127.0.0.1',
                'master_port' => 6379,
              },
              {'master_name'  => 'redis2',
                'master_ip'   => '127.0.0.1',
                'master_port' => 6380,
              },
          ],
        },
        # Secondary Sentinel
        {
          'sentinel_port' => 26380,
          'name' => '26380',
          'monitor' => [
              {'master_name'  => 'redis1',
                'master_ip'   => '127.0.0.1',
                'master_port' => 6379,
              },
              {'master_name'  => 'redis2',
                'master_ip'   => '127.0.0.1',
                'master_port' => 6380,
              },
          ],
        },
      ]

Sample config output:

    sentinel monitor redis1 127.0.0.1 6379 2
    sentinel down-after-milliseconds redis1 30000
    sentinel can-failover redis1 yes
    sentinel parallel-syncs redis1 1
    sentinel failover-timeout redis1 900000
  
    sentinel monitor redis2 127.0.0.1 6380 2
    sentinel down-after-milliseconds redis2 30000
    sentinel can-failover redis2 yes
    sentinel parallel-syncs redis2 1
    sentinel failover-timeout redis2 900000

Thoughts?

Kevin

### PR DESCRIPTION
**Pull request by martinb3**
- Add support in sentinel resource for an array of masters to monitor, with backwards compatibility for the older attributes, fixes #73. Replaces #87.
- Introduce a test-kitchen test for sentinel watching multiple masters.
- Incidentally, fixes #193 as well, since it adds a master name attribute for each master.

### CODE REVIEW

**Comment by martinb3** [comment]
@zarry or @brianbianco, I'd love to get your feedback on this. Thanks!

**Comment by zarry** [comment]
Sorry, been a bit busy lately.  I can likely make some time for this tonight.  If you don't hear back from me on this ticket tonight ping me again tomorrow.

File: templates/default/sentinel.conf.erb [hunk 1]
```diff
@@ -15,6 +15,8 @@ syslog-facility <%= @syslogfacility %>
 # The port that this sentinel instance will run on
 port <%=@sentinel_port%>
 
+<% @masters.each do |current| %>
```
Comment (zarry) [inline templates/default/sentinel.conf.erb]: What do you think about including the comments in each iteration?  Would this make the property file harder to read? Realistically does the readability even matter?

**Comment by zarry** [comment]
Love the test coverage! This looks great! I don't feel strongly enough about the one comment I had that it should hold up a merge anyway. Feel free to merge this whenever you are ready.

File: templates/default/sentinel.conf.erb (hunk 1 above)
Comment (martinb3) [inline templates/default/sentinel.conf.erb]: I think you're probably right. It does end up being a huge number of lines of just comments. I'll trim it down and push one more commit.

**Comment by martinb3** [comment]
Thanks @zarry! I added separate loops to reduce the duplicate comments.

**Pull request merged**